### PR TITLE
[PM-7907] Encourage The Use of UserId in CryptoService

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Various tech debt things where `CryptoService` should highly encourage the use of `UserId`, especially for mutations. Many of these uses are in either `SyncService` where the user id is easily available and during the login process where it can be made available with minor changes. 

This also makes the `SymmetricCryptoKey` parameter required in `makeKeyPair`, it updates the two call sites that weren't passing it in with either a `UserKey` that was already in context, and was very recently set or I called the method that `makeKeyPair` used to call for them.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by jlf0dev — APPROVED** [review]

File: libs/common/src/platform/abstractions/crypto.service.ts [hunk 1]
```diff
@@ -200,7 +212,13 @@ export abstract class CryptoService {
    * @param providers The providers to set keys for
    */
   abstract activeUserProviderKeys$: Observable<Record<ProviderId, ProviderKey>>;
-  abstract setProviderKeys(orgs: ProfileProviderResponse[]): Promise<void>;
+
+  /**
+   * Stores the provider keys for a given user.
+   * @param orgs The provider orgs for which to save the keys from.
+   * @param userId The user id of the user for which to store the keys for.
```
Comment (MGibson1) [inline libs/common/src/platform/abstractions/crypto.service.ts:219]: Please mention that this throws for non-guid userIds

File: libs/common/src/platform/abstractions/crypto.service.ts [hunk 2]
```diff
@@ -228,7 +246,7 @@ export abstract class CryptoService {
    * Note: does not clear the private key if null is provided
    * @param encPrivateKey An encrypted private key
    */
-  abstract setPrivateKey(encPrivateKey: string): Promise<void>;
+  abstract setPrivateKey(encPrivateKey: string, userId: UserId): Promise<void>;
```
Comment (MGibson1) [inline libs/common/src/platform/abstractions/crypto.service.ts:249]: Please mention that this throws for non-guid userIds

**Review by MGibson1 — COMMENTED** [review]
Two doc improvements. Really these throw because `getUser` throws, which is because build key for `UserKeyDefinition`s throws, it would be nice to be able to propagate these, but for now the best we can do is replicate in docs.

**Review by Jingo88 — APPROVED** [review]

File: libs/common/src/platform/abstractions/crypto.service.ts (hunk 1 above)
Comment (justindbaur) [reply libs/common/src/platform/abstractions/crypto.service.ts:219]: In my opinion that kind of throws doc doesn't really add much, once we get to a place where barely anybody is `as UserId`-ing that is like saying `@throws If you pass in a string to my `number` parameter`. The fact that it has its own type and `as UserId` should be considered unsafe means the compiler error of trying to pass in a non `UserId` is more than enough.

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**